src/private/unblocker.rs

```diff
@@ -73,10 +79,22 @@ where
     }
 
     extern "C" fn work(arg: *mut core::ffi::c_void) {
-        let worker: Box<Box<dyn FnOnce() + Send + 'static>> =
-            unsafe { Box::from_raw(arg as *mut _) };
+        {
+            let worker: Box<Box<dyn FnOnce() + Send + 'static>> =
+                unsafe { Box::from_raw(arg as *mut _) };
 
-        worker();
+            worker();
+        }
+        unsafe {
+            // FreeRTOS tasks must delete themselves. Returning from the task function
+            // without deleting the task will cause a crash.
+            // This function will immediately stop running. This means we have to ensure
+            // that all memory is freed before calling vTaskDelete.
+            // Fortunately, this function does not manage any memory at this point.
+            // The scope above guarantees that all relevant variables are dropped at this point.
+            // See https://www.freertos.org/implementing-a-FreeRTOS-task.html
+            task::destroy(std::ptr::null_mut());
```

Comment:
replace `std` with `core` please. That will also make CI happy.

Reply:
Fixed